charts/redpanda/templates/service.internal.yaml

```diff
@@ -16,14 +16,15 @@ limitations under the License.
 */}}
 ---
 # This service is only used to create the DNS enteries for each pod in
-# the stateful set. This service should not be used by any client
-# application
+# the stateful set and allow the serviceMonitor to target the pods.
+# This service should not be used by any client application
 apiVersion: v1
 kind: Service
 metadata:
   name: {{ include "redpanda.fullname" . }}
   namespace: {{ .Release.Namespace | quote }}
   labels:
+    prometheus/scrape: "true"
```

Comment:
should this be gated on monitoring.enabled?

Reply:
Is this even needed if you create a ServiceMonitor? iirc, you don't need both.

Reply:
I actually set this specific label so we can differenciate between this service and the others services from the chart, but sure I can gate it

Reply:
The `serviceMonitor` needs to select a `Service` based on labels. So I need to be able to pick the right one between the `service.internal`, `service.loadbalancer` and `service.nodeport` as they all show the same list of labels...

Reply:
Got it! Thanks.